libs/@local/harpc/server/src/lib.rs

```diff
@@ -7,4 +7,121 @@ pub mod error;
 pub mod route;
 pub mod router;
 pub mod serve;
+
 pub mod session;
+
+use core::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use error_stack::{Report, ResultExt};
+use futures::{Stream, StreamExt, stream::FusedStream};
+use harpc_codec::encode::ErrorEncoder;
+use harpc_net::{
+    session::server::{EventStream, ListenStream, SessionConfig, SessionLayer, Transaction},
+    transport::{TransportConfig, TransportLayer},
+};
+use multiaddr::Multiaddr;
+use tokio_util::sync::{CancellationToken, DropGuard};
+
+#[derive(
+    Debug,
+    Copy,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    derive_more::Display,
+    derive_more::Error,
```

Comment:
What is the reason to not use thiserror?

Reply:
`derive_more` was already a dependency. So I thought instead of pulling in `thiserror` into server, I'd just use that instead.

That raises a larger question - that I will revisit at some point - I've been increasingly dissatisfied with `thiserror` (`generic_member_access` not being a thing, wanting to derive `Display` - just like `Debug` - outside of error) to see where to go next. I personally like that `derive_more` is more of a compositional approach, but it doesn't implement `generic_member_access` (tho I personally feel like it'd be easier to contribute to `derive_more` than `thiserror`, than again - we have the proposal of #5352 with an example macro that might be separate from those two)

Reply:
I absolutely hear you and deriving `Error` for a type manually when `Display` is implemented is quite easy as well (without `generic_member_access` it's a one-liner). The hard thing about `Error` _is_ `Display`. Personally, I would probably tend to use `derive_more` more generally over `thiserror` because of it's separation of concerns as it only derives one trait per statement.